microbench/src/main/java/io/netty/microbench/channel/AutoFlushBenchmark.java

```diff
@@ -0,0 +1,242 @@
+        }
+        if (serverGroup != null) {
+            serverGroup.sync();
+        }
+        if (clientGroup != null) {
+            clientGroup.sync();
+        }
+    }
+
+    @Benchmark
+    public void measureWritesWithFlushOnEach() throws InterruptedException {
+        ChannelFuture lastWriteFuture = clientChannel.voidPromise();
+        if (flush) {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.writeAndFlush(payload.retain());
+            }
+        } else {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+            }
+        }
+
+        lastWriteFuture.sync().await(10, SECONDS);
+    }
+
+    @Benchmark
+    public void measureWritesWithFlushAtEnd() throws InterruptedException {
+        ChannelFuture lastWriteFuture = clientChannel.voidPromise();
+        if (flush) {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+            }
+            pipeline.flush();
+        } else {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+            }
+        }
+
+        lastWriteFuture.sync().await(10, SECONDS);
+    }
+
+    @Benchmark
+    public void measureWritesWithFlushEvery5() throws InterruptedException {
+        ChannelFuture lastWriteFuture = clientChannel.voidPromise();
+        if (flush) {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+                if (i % 5 == 0) {
+                    pipeline.flush();
+                }
+            }
+            pipeline.flush();
+        } else {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+            }
+        }
+
+        lastWriteFuture.sync().await(10, SECONDS);
+    }
+
+    @Benchmark
+    public void measureWritesWithFlushEverySecond() throws InterruptedException {
+        ChannelFuture lastWriteFuture = clientChannel.voidPromise();
+        if (flush) {
+            pipeline.channel().eventLoop().scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    pipeline.channel().flush();
+                }
+            }, 100, 100, SECONDS);
+
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+            }
+            pipeline.flush();
+        } else {
+            for (int i = 0; i < writeCount; i++) {
+                lastWriteFuture = pipeline.write(payload.retain());
+            }
+        }
+
+        lastWriteFuture.sync().await(10, SECONDS);
+    }
+
+    private static ByteBuf createData(int length) {
+        byte[] result = new byte[length];
+        new Random().nextBytes(result);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Random object created and used only once in io.netty.microbench.channel.AutoFlushBenchmark.createData(int) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ADMI_RANDOM_USED_ONLY_ONCE)